sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -11,256 +11,275 @@
 
 #ifdef __cplusplus
 extern "C" {
 #endif  // __cplusplus
 
 /**
- * Hardware instantiation parameters for SRAM Controller.
+ * A toggle state: enabled, or disabled.
  *
- * This struct describes information about the underlying hardware that is
- * not determined until the hardware design is used as part of a top-level
- * design.
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
  */
-typedef struct dif_sram_ctrl_params {
+typedef enum dif_sram_ctrl_toggle {
+  /**
+   * The "enabled" state.
+   */
+  kDifSramCtrlToggleEnabled,
   /**
-   * The base address for the SRAM Controller hardware registers.
+   * The "disabled" state.
    */
-  mmio_region_t base_addr;
-} dif_sram_ctrl_params_t;
-
-/**
- * A handle to SRAM Controller.
- *
- * This type should be treated as opaque by users.
- */
-typedef struct dif_sram_ctrl {
-  dif_sram_ctrl_params_t params;
-} dif_sram_ctrl_t;
+  kDifSramCtrlToggleDisabled,
+} dif_sram_ctrl_toggle_t;
 
 /**
- * The result of a SRAM Controller operation.
+ * SRAM Controller status.
  */
-typedef enum dif_sram_ctrl_result {
+typedef enum dif_sram_ctrl_status {
   /**
-   * Indicates that the operation succeeded.
+   * Bus integrity fault is detected. This error triggers a fatal_error alert.
+   * This condition is terminal.
    */
-  kDifSramCtrlOk = 0,
+  kDifSramCtrlStatusBusIntegErr = 0,
   /**
-   * Indicates some unspecified failure.
+   * Initialization counter has reached an invalid state. This error triggers
+   * a fatal_error alert. This condition is terminal.
    */
-  kDifSramCtrlError,
+  kDifSramCtrlStatusInitErr,
   /**
-   * Indicates that some parameter passed into a function failed a
-   * precondition.
-   *
-   * When this value is returned, no hardware operations occurred.
+   * SRAM Controller has received an escalate request, the scrambling keys have
+   * been reset to the default values and all subsequent memory requests will
+   * be blocked. This condition is terminal.
    */
-  kDifSramCtrlBadArg,
-} dif_sram_ctrl_result_t;
+  kDifSramCtrlStatusEscalated,
+  /**
+   * New scrambling key has been successfully obtained from OTP. If the flag is
+   * not set, the SRAM contents are still scrambled, but a default all-zero key
+   * and nonce are used to do so.
+   */
+  kDifSramCtrlStatusScrKeyValid,
+  /**
+   * Scrambling key has been derived from a valid key seed in OTP. When
+   * `kDifSramCtrlStatusScrKeyValid` is set, but this flag is unset - the
+   * scrambling key is still ephemeral (i.e., it is derived using entropy from
+   * CSRNG), but a default all-zero value is used as the key seed. This could
+   * happen when the scrambling key seeds have not yet been provisioned to OTP.
+   */
+  kDifSramCtrlStatusScrKeySeedValid,
+  /**
+   * Hardware initialization triggered via `dif_sram_ctrl_scramble` or
+   * `dif_sram_ctrl_wipe` has completed.
+   */
+  kDifSramCtrlStatusInitDone,
+} dif_sram_ctrl_status_t;
```

Comment:
IIRC the SRAM controller can be in more than one of these states at once. Might be worth using a bitset or struct of bools to represent the state.

Reply:
Taking `dif_rst_mgr` approach with returning 32bit register contents, and then using status bit-fields to extract individual/combined bits.

Done.

Reply:
These look off by 1 (`kDifSramCtrlStatusInitErr` should be `1 << 1` for example) and `kDifSramCtrlStatusInitDone` doesn't have a value set.

Reply:
(unresolving this so it isn't missed)